packages/backend/src/app/app-connection/app-connection.service.ts

```diff
@@ -189,6 +186,16 @@ const validateConnectionValue = async (params: ValidateConnectionValueParams): P
     return connection.value
 }
 
+function decryptConnection(encryptedConnection: AppConnectionSchema): AppConnection {
+    const connection = {
+        ...encryptedConnection,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        value: decryptObject<any>(encryptedConnection.value),
+    }
+    connection.status = getStatus(connection.value)
+    return connection
+}
```

Comment:
This bit is very confusing, `decryptConnection` returns an `AppConnection` so `connection` type is `AppConnection`, and `getStatus` takes an `AppConnection` so `connection.value` type is also `AppConnection`?!

I think this issue happened in the first place because of the wrong types so let's get types right this time to avoid future pitfalls, what do you think?